xarray/core/dataset.py

```diff
@@ -6410,8 +6432,15 @@ def to_dask_dataframe(
             if isinstance(var, IndexVariable):
                 var = var.to_base_variable()
 
+            # Make sure var is a dask array, otherwise the array can become too large
+            # when it is broadcasted to several dimensions:
+            if var.chunks is None:
+                var = var.chunk()
+
             dask_array = var.set_dims(ordered_dims).chunk(self.chunks).data
-            series = dd.from_array(dask_array.reshape(-1), columns=[name])
+            dask_array_raveled = ravel_chunks(dask_array)
```

Comment:
```suggestion
```

Unfortunately we can't do this, at least not by default. 

We could ask dask to add this behaviour as an opt-in kwarg for `dask.dataframe.from_array`

Reply:
How come?

If we go back to using `.reshape(-1)` or `.ravel()` we will continue getting this warning:
```
PerformanceWarning: Reshaping is producing a large chunk. To accept the large
chunk and silence this warning, set the option
with dask.config.set(**{'array.slicing.split_large_chunks': False}):
    array.reshape(shape)

To avoid creating the large chunks, set the option
with dask.config.set(**{'array.slicing.split_large_chunks': True}):
    array.reshape(shape)Explictly passing ``limit`` to ``reshape`` will also silence this warning
array.reshape(shape, limit='128 MiB')
  exec_fun(compile(ast_code, filename, 'exec'), ns_globals, ns_locals)
```

Reply:
reshape/ravel have an implied order. With this change the ordering of rows in the output dataframe depends on the chunking of the input array, which would be confusing as default behaviour

I think the warning is fine. Users can override with the dask context manager as suggested in the warning.

Reply:
Ok, I'll undo it, it wasn't necessary for the real fix anyway.

For comparison, here's the `df.visualization()`:

With reshape:
![image](https://user-images.githubusercontent.com/14371165/214383379-24877493-28b1-4be8-b415-4abbdffd14e1.png)

with reshape with context {'array.slicing.split_large_chunks': True}:
![image](https://user-images.githubusercontent.com/14371165/214396092-48056d9c-49f4-4559-bcbe-b5df69f3d49e.png)


With ravel_chunks:
![image](https://user-images.githubusercontent.com/14371165/214383914-65d7513b-0423-48de-8809-7ae5554c3ee7.png)

Reply:
Yup, it's a big improvement. If you're dying to add it someplace, [polyfit](https://github.com/pydata/xarray/issues/5629#issuecomment-960133879) would be a good candidate (and very impactful PR).